src/templates.rs

```diff
@@ -339,6 +367,41 @@ pub static TEMPLATE_BWD_COMPILATION_METRICS: &str = r#"
     {{ else }}
     <p> No failures! </p>
     {{ endif }}
+    "#,
+    TEMPLATE_QUERY_PARAM_SCRIPT,
+    r#"
 </body>
 </html>
-"#;
+"#
+);
+
+pub const TEMPLATE_QUERY_PARAM_SCRIPT: &str = r#"
+    <script>
+    document.addEventListener('DOMContentLoaded', function() \{
+
+        // Append the current URL's query parameters to all relative links on the page
+        const queryParams = new URLSearchParams(window.location.search);
+        function appendQueryParams(url) \{
+            if (queryParams.size === 0) return url; // No query params, return original URL
+
+            const parsedUrl = new URL(url, window.location.href);
+            const updatedSearchParams = new URLSearchParams(parsedUrl.search);
+
+            // Append query parameters
+            for (const [key, value] of queryParams) \{
+                updatedSearchParams.set(key, value);
+            }
+
+            parsedUrl.search = updatedSearchParams.toString();
+            return parsedUrl.href;
+        }
+
+        // Select all relative links on the page
+        const relativeLinks = document.querySelectorAll('a[href]:not([href^="http://"]):not([href^="https://"]):not([href^="\#"])');
```

Comment:
Instead of a ton of selectors here you could just test if you have a relative URL on parsedUrl inside appendQueryParams?

Reply:
I could, but this is cleaner IMO since we can check that all the correct elements are being selected, and we need some selector anyway to get the `<a>` elements. If you want me to loop through all the selectors and test that's fine too.